### ISSUE TITLE
Enable rotation in isct_antsSliceRegularizedRegistration

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#1178]
ANTs needs to be recompiled and binaries need to be updated for OSX and Linux.

**Comment by jcohenadad** [comment]
there is a bug in the latest version of `antsSliceRegularizedRegistration` and B. Avants is unresponsive so we keep the label "wontfix" from the time being.

**Comment by joshuacwnewton** [comment]
I believe isct_antsSliceRegularizedRegistration has since been recompiled. See #2821.

As for enabling rotation, the proposed changes to `sct_register_multimodal` were here: https://github.com/neuropoly/spinalcordtoolbox/pull/1179/files#r505706963

It looks like this change was indeed applied. See below for the matching section of the current codebase:

https://github.com/neuropoly/spinalcordtoolbox/blob/cab82fd13425515a8568225dbe7fc003d9be2b1f/spinalcordtoolbox/registration/register.py#L160-L172

So, I consider this issue to have been addressed. Closing.

**Comment by joshuacwnewton** [comment]
Oops. My apologies, I misread. The fix has _not_ been reapplied. Reopening -- let me rewrite.

**Comment by joshuacwnewton** [comment]
Alright, I found the [source code for antsSliceRegularizedRegistration](https://github.com/benjamindeleener/ANTs/blob/master/Examples/antsSliceRegularizedRegistration.cxx). There is only the `translation` option:

https://github.com/benjamindeleener/ANTs/blob/51b5dfb6a0c8f9b573049e576142caead0530a2d/Examples/antsSliceRegularizedRegistration.cxx#L702

All other transform options (including `Rigid`) fail:

https://github.com/benjamindeleener/ANTs/blob/51b5dfb6a0c8f9b573049e576142caead0530a2d/Examples/antsSliceRegularizedRegistration.cxx#L741-L745

Because the rotation feature is not available (even after several years), I'm closing this issue.

**Comment by jcohenadad** [comment]
Reopening this issue because feature request was made by a [user](https://forum.spinalcordmri.org/t/2-questions-sct-fmri-moco-and-sct-extract-metric/628/3?u=jcohenadad).

**Comment by joshuacwnewton** [comment]
Just to clarify, @jcohenadad, I had closed this issue because when I last checked (https://github.com/neuropoly/spinalcordtoolbox/issues/1178#issuecomment-709513050), the upstream ANTs C source code for `antsSliceRegualizedRegistration` did not support `rotation` as an option.

Though, I am a bit confused as to why we thought `rotation` was an available option back in 2017 when this issue was first opened. Perhaps I missed something?

**Comment by jcohenadad** [comment]
Good point, I should have clarified: the issue implies reaching out to the ANTs team to enable rotation (or to do it ourself).

**Comment by joshuacwnewton** [comment]
Ahh, thank you for the clarification. I'll update the issue description.

**Comment by joshuacwnewton** [comment]
In [my earlier comment](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/1178#issuecomment-709513050), I said that "`Rigid`" was not supported. But, I had made a mistake: I was  looking at @benjamindeleener's fork of ANTs, which was last updated in 2015. :fearful:  

Looking at the actual, up-to-date source code of ANTs, they [do indeed appear to support rotation](https://github.com/ANTsX/ANTs/blob/a7eca60da98b311869a9ff24f52c7b3c819aa76e/Examples/antsSliceRegularizedRegistration.cxx#L741-L744). So, we should be able to enable this feature.

**Comment by jcohenadad** [comment]
> In [my earlier comment](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/1178#issuecomment-709513050), I said that "`Rigid`" was not supported. But, I had made a mistake: I was looking at @benjamindeleener's fork of ANTs, which was last updated in 2015. (My apologies for this... 😨)
> 
> Looking at the actual, up-to-date source code of ANTs, they [do indeed appear to support rotation](https://github.com/ANTsX/ANTs/blob/a7eca60da98b311869a9ff24f52c7b3c819aa76e/Examples/antsSliceRegularizedRegistration.cxx#L741-L744). So, (hopefully?) we should be able to enable this feature.

if the ["bug"](https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/1178#issuecomment-387598325) was fixed, then yes we should definitely enable this. sorry i should have documented the bug better-- IIRC the problem was that no rotation was properly estimated. Here is my latest email conversation with developer Brian Avants:

> I’ve been playing around with antsSliceRegularizedRegistration and I noticed that the X_TxTy_poly.csv output does not correspond to the XWarp.nii.gz. While the first is regularized as expected, the warping field is not.
Is this a bug, or something I am missing in the function call?
Example of call:
`antsSliceRegularizedRegistration -t Rigid[1] -m CC[dest.nii,src.nii,1,4,Regular,0.2] -p 2x2x1 -i 10 -f 2 -s 0 -v 1 -o [step1,src_reg_crop_regStep1.nii] -x mask.nii.gz`

**Comment by joshuacwnewton** [comment]
> IIRC the problem was that no rotation was properly estimated. Here is my latest email conversation with developer Brian Avants:

Aha, I see. Thank you for the clarification. 

In that case, we might want to consider opening an issue about this upstream to have this behavior documented publicly. (I did try searching the ANTs repo for similar reports, but IIRC I couldn't find anything.)
